lib/env.go

```diff
@@ -0,0 +1,22 @@
+package lib
+
+import "strings"
+
+// ParseEnvKeyValue splits an environment variable string into key and value.
+func ParseEnvKeyValue(kv string) (string, string) {
+	if idx := strings.IndexRune(kv, '='); idx != -1 {
+		return kv[:idx], kv[idx+1:]
+	}
+	return kv, ""
+}
+
+// BuildEnvMap returns a map from raw environment values, such as returned from
+// os.Environ().
+func BuildEnvMap(environ []string) map[string]string {
+	env := make(map[string]string, len(environ))
+	for _, kv := range environ {
+		k, v := ParseEnvKeyValue(kv)
+		env[k] = v
+	}
+	return env
+}
```

Comment:
Moving these here doesn't seem like an improvement. 

First, we are trying to reduce the size of `lib/`, since it [itself is an antipattern](https://go.dev/blog/package-names), not increase it.

But, more importantly, these functions are not widely used enough to move them here. `ParseEnvKeyValue()` is only used here, in `BuildEnvMap()`, and in a single other place in `cmd/`. `BuildEnvMap()` itself is only used in `cmd/state` and in a test (where it's a convenience. not a necessity).

Reply:
I just moved them to `cmd/state` in 504f134a. Let me know if you have a better idea.

Reply:
Still think they should probably be unexported and copied, but the bigger issue was having them in `lib/`, so I am reasonably happy with this compromise :sweat_smile: